pkg/trace/stats/aggregation.go

```diff
@@ -62,9 +64,9 @@ func getStatusCode(s *pb.Span) uint32 {
 }
 
 // NewAggregationFromSpan creates a new aggregation from the provided span and env
-func NewAggregationFromSpan(s *pb.Span, origin string, aggKey PayloadAggregationKey) Aggregation {
+func NewAggregationFromSpan(s *pb.Span, origin string, aggKey PayloadAggregationKey, extraAggregators map[string]struct{}) Aggregation {
```

Comment:
Why do we need this `extraAggregators` map? When the `PayloadAggregationKey` is constructed, it should optionally set the `PeerService` value, no?